src/drivers/ata.c

```diff
@@ -234,12 +213,43 @@ void ata_io_cmd(void * _dev, int cmd, void * buf, range blocks, status_handler s
         ata_out8(dev, ATA_SECTOR, lba);
         ata_out8(dev, ATA_DRIVE, ATA_D_LBA | ATA_DEV(dev->unit));
     } else {
+        assert(nsectors <= 255);
         ata_out8(dev, ATA_COUNT, nsectors);
         ata_out8(dev, ATA_CYL_MSB, lba >> 16);
         ata_out8(dev, ATA_CYL_LSB, lba >> 8);
         ata_out8(dev, ATA_SECTOR, lba);
         ata_out8(dev, ATA_DRIVE, ATA_D_IBM | ATA_D_LBA | ATA_DEV(dev->unit) | ((lba >> 24) & 0x0f));
     }
+    return true;
+}
+
+void ata_io_cmd(void * _dev, int cmd, void * buf, range blocks, status_handler s)
+{
+    struct ata *dev = (struct ata *)_dev;
+    const char *err;
+
+    u64 lba = blocks.start;
+    u64 nsectors = range_span(blocks);
+    if (nsectors == 0) {
+        const char *err = "ata_io_cmd: zero blocks I/O";
+        apply(s, timm("result", "%s", err));
```

Comment:
This code was already present before this PR, it might look as new code because of the refactoring I did to add the ata_set_lba() function.
Anyway, we are not using apply_status() anymore, and I now put the message in the timm string literal.